docs/cudf/source/errors.md

```diff
@@ -0,0 +1,46 @@
+# cuDF Error Handling Documentation (draft)
+
+## Scope
+
+This document serves to guide the use of error types when dealing with different exceptions encountered within cuDF domain. As of the draft version, it also serves as a reference for supporting libraries (like libcudf) to boost awareness of common error types encountered and to standardize their error interface. Note that this document will not serve as a complete list of error types thrown by cuDF.
+
+## Overview
+
+On a high level, exceptions consist of exception type and exception payload. Each can be standardized by the API. cuDF only standardizes exception type and uses the payload to convey debugging information for users. (Subject to discussion: is it sufficient to only use builtin error types but not the payload to signal all errors?)
```

Comment:
A related question: since we aim to match pandas API, do we consider it sufficient to match API up to the exception type alone, or do we also need to match the message? I think type is sufficient, but we should standardize around that.

Reply:
I'm not sure I understand exactly what your discussion question is asking. Are you asking whether the payload should be promised by the API as well?

Reply:
@vyasr yes that's the question: should the API also promise the error payload? Take `ValueError` for example, the semantic could vary even for the same API: e.g.
https://github.com/rapidsai/cudf/blob/4893259b2fd6cf1f2079eff68249290708519892/python/cudf/cudf/core/column/numerical.py#L564
https://github.com/rapidsai/cudf/blob/4893259b2fd6cf1f2079eff68249290708519892/python/cudf/cudf/core/column/numerical.py#L580

Reply:
I think promising a consistent payload is a nice-to-have, but not a must. In an ideal world, yes it should be consistent, but given the current maturity of the project and the fast evolution it is probably too much to ask. I'd say we could reassess this if/when the data APIs are mature enough that we can decouple somewhat from pandas, but for now our decision should be governed by whatever pandas promises. @shwina was fine with having our exception deviate from pandas in some recent PR of mine that he approved, so he may also have thoughts here.